src/schedule.cpp

```diff
@@ -304,6 +304,52 @@ struct stream_info
         return result;
     }
 
+    void
+    compute_conflict_table(instruction_ref ins,
+                           const std::vector<std::vector<instruction_ref>>& merge_ins,
+                           std::unordered_map<instruction_ref, std::unordered_set<instruction_ref>>&
+                               thrd_conflict_table)
+    {
+        int size = static_cast<int>(merge_ins.size());
+        std::unordered_set<instruction_ref> checked_ins1_set;
+        for(int i = 0; i < size - 1; ++i)
+        {
+            std::unordered_set<instruction_ref> ins1_set;
+            for(auto ins1 : merge_ins[i])
+            {
+                if(checked_ins1_set.count(ins1) == 0)
+                    ins1_set.insert(ins1);
+            }
+            checked_ins1_set.insert(ins1_set.begin(), ins1_set.end());
+
+            std::unordered_set<instruction_ref> ins2_set;
+            for(auto j = i + 1; j < size; ++j)
+            {
+                for(auto ins2 : merge_ins[j])
+                {
+                    if(checked_ins1_set.count(ins2) == 0)
+                        ins2_set.insert(ins2);
+                }
+            }
+
+            for(auto ins1 : ins1_set)
+            {
+                auto p1 = std::distance(ins1, ins);
```

Comment:
How do we know that `ins1` always precedes `ins`?

Reply:
instructions in the concurrent table meets that requirement